### PR TITLE
chore: enhance websocket test-server capabilities

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: src/Playwright.Tests/BaseTests/PlaywrightAssert.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ internal static class PlaywrightAssert
             try
             {
                 await action();
-                Assert.Fail();
+                Assert.Fail($"Expected exception of type '{typeof(T).Name}' which was not thrown");
```
Comment (mxschmitt) [inline src/Playwright.Tests/BaseTests/PlaywrightAssert.cs:42]: drive-by: better exception if the exception does not get thrown

File: src/Playwright/Transport/Channels/BrowserContextChannel.cs [hunk 2]
```diff
@@ -165,15 +165,6 @@ internal Task SetNetworkInterceptionEnabledAsync(bool enabled)
                     ["enabled"] = enabled,
                 });
 
-        internal Task SetHttpCredentialsAsync(HttpCredentials credentials)
```
Comment (mxschmitt) [inline src/Playwright/Transport/Channels/BrowserContextChannel.cs:168]: drive-by: not used

**Review by pavelfeldman — APPROVED** [review]
